source/slang/hlsl.meta.slang

```diff
@@ -11406,15 +11406,22 @@ ${{{{
 //
 // Vulkan/SPIR-V specific features
 //
-
-struct VkSubpassInput<T>
+[sealed]
+[builtin]
+interface __ITextureSubpass
 {
-    T SubpassLoad();
+    static const int flavor;
 }
-
-struct VkSubpassInputMS<T>
+__magic_type(TextureShapeSubpassType)
+__intrinsic_type($(kIROp_TextureShapeSubpassType))
+struct __ShapeSubpass : __ITextureSubpass
+{
+    static const int flavor = $(SLANG_TEXTURE_SUBPASS);
+}
+__magic_type(TextureType)
+__intrinsic_type($(kIROp_TextureType))
```

Comment:
Why does it need to be TextureType?

Can we use a separate type for ast and IR, so we can get rid of this unnecessary generic arguments that are irrelevant to this type.

That will also allow us to get rid of the ITextureShapeSubpass interface.